### ISSUE TITLE
3.18 - Preconnect and prefetch removal

### ISSUE THREAD

**Issue by piotrbak** [wp-media/wp-rocket#7066]
**User Story**
As a user, I want to remove preconnect/prefetch links to Google Fonts when locally hosted fonts are served

**Acceptance Criteria**
- The feature will remove all preconnect/prefetch links related to the Google Fonts
 `fonts.gstatic.com` delivers the font files
`fonts.googleapis.com` delivers font CSS
- We'll introduce a filter to disable this behaviour

**Comment by Khadreal** [comment]
This should be pick up after #7063 
# scope a solution:
The process responsible for the frontend rewrite, we should have a variable that is set to true if locally hosted fonts are served. 
If the variable is true we call the `remove_preconnect_and_prefetch` method, in this method we check if the filter is not set to false.

```
private function remove_preconnect_and_prefetch( string $html ) {
    if (!$html) {
        return $html;        
    }
    $remove_links = wpm_apply_filter( 'boolean', 'rocket_remove_font_pre_links', true );

   if( ! $remove_links ) {
     return $html;
    }

    $pattern = '/<link(?:[^>]*)(?:rel=["\'](?:dns-prefetch|preconnect)["\'])(?:[^>]*)(?:href=["\'](?:https?:)?\/\/(?:fonts\.(?:googleapis|gstatic)\.com)["\'])(?:[^>]*)>/i';

    $html = preg_replace($pattern, '', $html);

    return $html;
}
```

We can also use the `wp_resource_hints` filter to make sure all preconnect/prefetch links are removed.

Estimation
[S]

**Comment by jeawhanlee** [comment]
LGTM

**Comment by remyperona** [comment]
What should happen in the case we have exclusions? Should the links be preserved?

**Comment by piotrbak** [comment]
Good point @remyperona 

Yes

**Comment by piotrbak** [comment]
@remyperona Do you think it should be added here, or maybe as a separate issue, as it's not necessary for the initial product

**Comment by remyperona** [comment]
It could be part of the exclusions user story
